### ISSUE TITLE
Trying to get property of non-object $fieldSettings

### ISSUE THREAD

**Issue by themccallister** [engram-design/Comments#29]
In `CommentsController > actionAdd` the following is returning null:

`$fieldSettings = craft()->comments_settings->getFieldSettings(craft()->request->getPost('elementId'));`

I did verify that the elementId is valid, I tried to wrap a conditional around the field settings for the controller like so:

```php
// check if field settings is null
if (!is_null($fieldSettings)) {
    // Set any new comment to be pending if requireModeration is true
    if ($fieldSettings->requireModeration) {
        $commentModel->status = Comments_CommentModel::PENDING;
    }

    // Protect against Anonymous submissions, if turned off
    if (!$fieldSettings->allowAnonymous && !$commentModel->userId) {
        craft()->comments->response($this, array('error' => 'Must be logged in to comment.'));
    }
} else {
    $commentModel->status = Comments_CommentModel::APPROVED;
}
```

This does solve the problem, I'm not sure that its optimal or meets your goals for comment_settings.

**Comment by svenio** [comment]
+1

**Comment by lesliedoherty** [comment]
Same. Will use the conditional wrapping for now.

**Comment by engram-design** [comment]
@themccallister This has been properly fixed in the [0.3.8](https://github.com/engram-design/Comments/releases/tag/0.3.8) release which fixes a lot of bugs, including this one. Feel free to reopen if you continue to have issues!
